middleware/proxy/google.go

```diff
@@ -218,23 +219,30 @@ func newUpstream(hosts []string, old *staticUpstream) Upstream {
 			Conns:       0,
 			Fails:       0,
 			FailTimeout: upstream.FailTimeout,
-			Unhealthy:   false,
+			OkUntil:     time.Now().Add(time.Duration(24 * 365 * 100 * time.Hour)), // now + 100 years
```

Comment:
OK.  I already use time.Unix(0,0) to signal a failed health check (because Now() is always after it).  So we'd make the check:
if ! uh.OkUntil.IsZero() && time.Now().After(uk.OkUntil) {
    // fail
}
?
That is different from time.Unix(0,0) so the mechanism would still work.